From c9015824b075be99bab6c57ae95682df52eb6e91 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Tue, 4 Dec 2018 21:06:33 +0300
Subject: [PATCH 0851/1051] soc: octeontx2: replace existing mbox req with
 lighter req message

Currently, sso_grp_priority is used for fetching priority info form SSO.
While requesting info from SSO we only need HWGRP/HWS slot id in the
request message, all the other fields are unused.
Add generic info mbox request message that can be used for requesting
GGRP/HWS info form SSO such as priority.

Change-Id: I49bb24bb23b2204a500dbc7eccbcbcbffba59224
Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1758
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/mbox.h    | 10 +++++++++-
 drivers/soc/marvell/octeontx2/rvu.h     |  2 +-
 drivers/soc/marvell/octeontx2/rvu_sso.c |  2 +-
 3 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/mbox.h b/drivers/soc/marvell/octeontx2/mbox.h
index 2f787364bdd8..e71c7aa69228 100644
--- a/drivers/soc/marvell/octeontx2/mbox.h
+++ b/drivers/soc/marvell/octeontx2/mbox.h
@@ -161,7 +161,7 @@ M(SSO_HW_SETCONFIG,	0x604, sso_hw_setconfig, sso_hw_setconfig, msg_rsp)\
 M(SSO_GRP_SET_PRIORITY,	0x605, sso_grp_set_priority,			\
 				sso_grp_priority, msg_rsp)		\
 M(SSO_GRP_GET_PRIORITY,	0x606, sso_grp_get_priority,			\
-				sso_grp_priority, sso_grp_priority)	\
+				sso_info_req, sso_grp_priority)		\
 M(SSO_WS_CACHE_INV,	0x607, sso_ws_cache_inv, msg_req, msg_rsp)	\
 M(SSO_GRP_QOS_CONFIG,	0x608, sso_grp_qos_config, sso_grp_qos_cfg, msg_rsp)\
 /* TIM mbox IDs (range 0x800 - 0x9FF) */				\
@@ -753,6 +753,14 @@ struct sso_hw_setconfig {
 	u16	hwgrps;
 };
 
+struct sso_info_req {
+	struct mbox_msghdr hdr;
+	union {
+		u16 grp;
+		u16 hws;
+	};
+};
+
 struct sso_grp_priority {
 	struct mbox_msghdr hdr;
 	u16 grp;
diff --git a/drivers/soc/marvell/octeontx2/rvu.h b/drivers/soc/marvell/octeontx2/rvu.h
index 8fd887ad5400..e4cb0d89078b 100644
--- a/drivers/soc/marvell/octeontx2/rvu.h
+++ b/drivers/soc/marvell/octeontx2/rvu.h
@@ -445,7 +445,7 @@ int rvu_mbox_handler_ssow_lf_free(struct rvu *rvu,
 				  struct ssow_lf_free_req *req,
 				  struct msg_rsp *rsp);
 int rvu_mbox_handler_sso_grp_get_priority(struct rvu *rvu,
-					  struct sso_grp_priority *req,
+					  struct sso_info_req *req,
 					  struct sso_grp_priority *rsp);
 int rvu_mbox_handler_sso_grp_set_priority(struct rvu *rvu,
 					  struct sso_grp_priority *req,
diff --git a/drivers/soc/marvell/octeontx2/rvu_sso.c b/drivers/soc/marvell/octeontx2/rvu_sso.c
index 44b8ed5c85fe..df88c75a0065 100644
--- a/drivers/soc/marvell/octeontx2/rvu_sso.c
+++ b/drivers/soc/marvell/octeontx2/rvu_sso.c
@@ -248,7 +248,7 @@ int rvu_mbox_handler_sso_grp_set_priority(struct rvu *rvu,
 }
 
 int rvu_mbox_handler_sso_grp_get_priority(struct rvu *rvu,
-					  struct sso_grp_priority *req,
+					  struct sso_info_req *req,
 					  struct sso_grp_priority *rsp)
 {
 	struct rvu_hwinfo *hw = rvu->hw;
-- 
2.17.1

